src/inversion_scripts/operators/TROPOMI_operator.py

```diff
@@ -143,31 +160,107 @@ def apply_average_tropomi_operator(
 
         # If building Jacobian matrix from GEOS-Chem perturbation simulation sensitivity data:
         if build_jacobian:
-            # Get GEOS-Chem perturbation sensitivities at this lat/lon, for all vertical levels and state vector elements
-            sensi_lonlat = GEOSCHEM["Sensitivities"][
-                gridcell_dict["iGC"], gridcell_dict["jGC"], :, :
-            ]
-            # Map the sensitivities to TROPOMI pressure levels
-            sat_deltaCH4 = remap_sensitivities(
-                sensi_lonlat,
-                merged["data_type"],
-                merged["p_merge"],
-                merged["edge_index"],
-                merged["first_gc_edge"],
-            )  # mixing ratio, unitless
-            # Tile the TROPOMI averaging kernel
-            avkern_tiled = np.transpose(np.tile(avkern, (n_elements, 1)))
-            # Tile the TROPOMI dry air subcolumns
-            dry_air_subcolumns_tiled = np.transpose(
-                np.tile(dry_air_subcolumns, (n_elements, 1))
-            )  # mol m-2
-            # Derive the change in column-averaged XCH4 that TROPOMI would see over this ground cell
-            jacobian_K[i, :] = np.sum(
-                avkern_tiled * sat_deltaCH4 * dry_air_subcolumns_tiled, 0
-            ) / sum(
-                dry_air_subcolumns
-            )  # mixing ratio, unitless
+            
+            # redundant code here...
+            
+            if config['OptimizeOH']:
+                vars_to_xch4 = ['jacobian_ch4', 'emis_base_ch4', 'oh_base_ch4']
+            else:
+                vars_to_xch4 = ['jacobian_ch4', 'emis_base_ch4']
+                
+            xch4 = {}
+                
+            for v in vars_to_xch4:
+                # Get GEOS-Chem jacobian ch4 at this lat/lon, for all vertical levels and state vector elements
+                jacobian_lonlat = GEOSCHEM[v][
+                    gridcell_dict["iGC"], gridcell_dict["jGC"], :, :
+                ]
+                # Map the sensitivities to TROPOMI pressure levels
+                sat_deltaCH4 = remap_sensitivities(
+                    jacobian_lonlat,
+                    merged["data_type"],
+                    merged["p_merge"],
+                    merged["edge_index"],
+                    merged["first_gc_edge"],
+                )  # mixing ratio, unitless
+                # Tile the TROPOMI averaging kernel
+                avkern_tiled = np.transpose(np.tile(avkern, (n_elements, 1)))
+                # Tile the TROPOMI dry air subcolumns
+                dry_air_subcolumns_tiled = np.transpose(
+                    np.tile(dry_air_subcolumns, (n_elements, 1))
+                )  # mol m-2
+                # Derive the change in column-averaged XCH4 that TROPOMI would see over this ground cell
+                #jacobian_K[i, :] = np.sum(
```

Comment:
remove this commented out code?

Reply:
Cleaned this up: https://github.com/geoschem/integrated_methane_inversion/pull/252/commits/07e05a19bd0e48c36442d712feb2567a033b3162